public/test/plugin.test.ts

```diff
@@ -0,0 +1,178 @@
+          clusterPermissions: { include: [] },
+          indexPermissions: { include: [] },
+          disabledTransportCategories: { exclude: [] },
+          disabledRestCategories: { exclude: [] },
+          ui: { autologout: false },
+        }),
+      },
+    };
+    deps = {
+      dataSource: { dataSourceEnabled: true },
+      savedObjectsManagement: { createSetup: jest.fn() },
+    };
+  });
+
+  it('does not call register function for certain applications when getNavGroupEnabled is off', async () => {
+    // Mock hasApiPermission to return false
+    pluginModule.hasApiPermission.mockResolvedValue(false); // Access the mock via the imported module
+
+    // Instantiate the plugin after mocking
+    plugin = new SecurityPlugin(initializerContext);
+
+    // Override getNavGroupEnabled to return false
+    coreSetup.chrome.navGroup = {
+      ...coreSetup.chrome.navGroup,
+      getNavGroupEnabled: () => false,
+    };
+    // Mock the core.application.register function
+    const registerSpy = jest.spyOn(coreSetup.application, 'register');
+
+    // Execute the setup function
+    await plugin.setup(coreSetup, deps);
+
+    // Assert that the register function was not called for specific applications
+    const registeredApps = registerSpy.mock.calls.map((call) => call[0].id);
+    const expectedApps = [
+      PLUGIN_GET_STARTED_APP_ID,
+      PLUGIN_AUTH_APP_ID,
+      PLUGIN_ROLES_APP_ID,
+      PLUGIN_USERS_APP_ID,
+      PLUGIN_PERMISSIONS_APP_ID,
+      PLUGIN_TENANTS_APP_ID,
+      PLUGIN_AUDITLOG_APP_ID,
+    ];
+
+    expectedApps.forEach((app) => {
+      expect(registeredApps).not.toContain(app);
+    });
+  });
+
+  it('calls register function for certain applications when getNavGroupEnabled is on', async () => {
+    // Mock hasApiPermission to return true
+    pluginModule.hasApiPermission.mockResolvedValue(true); // Access the mock via the imported module
+
+    // Instantiate the plugin after mocking
+    plugin = new SecurityPlugin(initializerContext);
+
+    // Override getNavGroupEnabled to return true
+    coreSetup.chrome.navGroup = {
+      ...coreSetup.chrome.navGroup,
+      getNavGroupEnabled: () => true,
+    };
+    // Mock the core.application.register function
+    const registerSpy = jest.spyOn(coreSetup.application, 'register');
+
+    // Execute the setup function
+    await plugin.setup(coreSetup, deps);
+
+    // Assert that the register function was called for specific applications
+    const registeredApps = registerSpy.mock.calls.map((call) => call[0].id);
+    const expectedApps = [
+      PLUGIN_GET_STARTED_APP_ID,
+      PLUGIN_AUTH_APP_ID,
+      PLUGIN_ROLES_APP_ID,
+      PLUGIN_USERS_APP_ID,
+      PLUGIN_PERMISSIONS_APP_ID,
+      PLUGIN_TENANTS_APP_ID,
+      PLUGIN_AUDITLOG_APP_ID,
+    ];
+
+    expectedApps.forEach((app) => {
+      expect(registeredApps).toContain(app);
+    });
+  });
+
+  it('does not call register function for tenant app when multitenancy is off', async () => {
```

Comment:
Should be covered in the test right above this, since that one shows that tenants is registered, and that has a different config where multitenancy is ON.